go-msgpack/codec/codecs_test.go

```diff
@@ -781,8 +782,13 @@ func doTestRpcOne(t *testing.T, rr Rpc, h Handle, doRequest bool, exitSleepMs ti
 				return // exit serverFn goroutine
 			}
 			if err1 == nil {
-				var sc rpc.ServerCodec = rr.ServerCodec(conn1, h)
-				srv.ServeCodec(sc)
+				var sc = rr.ServerCodec(conn1, h)
+				for {
+					if err := srv.ServeRequest(sc); err == io.EOF {
+						sc.Close()
+						break
+					}
+				}
```

Comment:
Once `ServeCodec` was refactor to use `ServeRequest` it became just these few lines, so I've inlined them into the two places it was called in tests.

The decision to include `go-msgpack` in this same fork is showing value here, because we can easily change these tests at the same time.